ts/WoltLabSuite/Core/Ui/Poll/Manager/Vote.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Handles the poll voting.
+ *
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2022 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Poll/Manager/Vote
+ * @since   5.5
+ */
+
+import AjaxRequest from "../../../Ajax/Request";
+import Manager, { PollViews } from "./Manager";
+import { ResponseData } from "../../../Ajax/Data";
+import * as Core from "../../../Core";
+
+export class Vote {
+  private pollManager: Manager;
+
+  private button: HTMLButtonElement;
+
+  private inputs: NodeListOf<HTMLInputElement>;
+
+  public constructor(manager: Manager) {
+    this.pollManager = manager;
+
+    this.initButton();
+    this.initSelects();
+  }
+
+  private initButton(): void {
+    const button = this.pollManager.getPollContainer().querySelector<HTMLButtonElement>(".votePollButton");
+
+    if (!button) {
+      throw new Error(`Could not find vote button for poll "${this.pollManager.pollID}".`);
+    }
+
+    this.button = button;
+
+    this.button.addEventListener("click", () => this.submit());
+  }
+
+  public initSelects(): void {
+    const container = this.pollManager.getPollContainer().querySelector(".pollVoteContainer");
+
+    if (container) {
+      this.inputs = container.querySelectorAll<HTMLInputElement>("input");
+
+      this.inputs.forEach((input) => {
+        input.addEventListener("change", () => this.checkInputs());
+      });
+
+      this.checkInputs();
+    }
+  }
+
+  private checkInputs(): void {
+    let selectedInputCount = 0;
+    this.inputs.forEach((input) => {
+      if (input.checked) {
+        selectedInputCount++;
+      }
+
+      if (this.pollManager.maxVotes > 1) {
+        input.disabled = false;
+      }
```

Comment:
This should be done exactly once during the initialization instead of with every invocation.